databroker/tests/test_broker.py

```diff
@@ -237,6 +239,29 @@ def test_indexing(db_empty, RE, hw):
         db[-11]
 
 
+@pytest.mark.parametrize(
+    "key, expected",
+    (
+        (np.int64(1), does_not_raise()),  # Key is a Scan ID
+        (np.int64(-1), does_not_raise()),  # Key is Nth-last scan
+        (np.int64(2**33), pytest.raises(KeyError)),  # Key is too large
```

Comment:
What about `-np.int64(2**33)`?

Reply:
Great suggestion!  I'll add it.

Reply:
Added this test case.  No changes to databroker code were needed.  Although as @tacaswell we should be able to bypass checking the size of the int-valued variable.